cloud/linode/loadbalancers.go

```diff
@@ -485,26 +483,24 @@ func getNodeInternalIP(node *v1.Node) string {
 	return ""
 }
 
-func getTLSCertInfo(kubeClient kubernetes.Interface, tlsAnnotations []*tlsAnnotation, namespace string, port int) (string, string, error) {
-	for _, tlsAnnotation := range tlsAnnotations {
-		if tlsAnnotation.Port == port {
-			secret, err := kubeClient.CoreV1().Secrets(namespace).Get(tlsAnnotation.TLSSecretName, metav1.GetOptions{})
-			if err != nil {
-				return "", "", err
-			}
+func getTLSCertInfo(kubeClient kubernetes.Interface, namespace string, portConfig *portConfig) (string, string, error) {
```

Comment:
As above, this can (and probably should) take a `portConfig`, not a `*portConfig`.